libbeat/cmd/instance/beat.go

```diff
@@ -388,7 +388,7 @@ func NewBeatReceiver(settings Settings, receiverConfig map[string]interface{}, c
 	}
 
 	// log paths values to help with troubleshooting
-	logp.Info(paths.Paths.String())
+	logp.Info("%s", paths.Paths.String())
```

Comment:
Why do we need this change? Why can't we just pass the string?

Reply:
Info calls Sprintf behind the scenes, according to the new go vet rule in 1.24 the first argument to a printf style function should be a static string with optional format specifiers. Since we were passing a dynamic string previously it complains about this code.

```
// Info uses fmt.Sprintf to construct and log a message.
// Deprecated: Use logp.NewLogger.
func Info(format string, v ...interface{}) {
	log := globalLogger()
	if log.Core().Enabled(zap.InfoLevel) {
		log.Info(fmt.Sprintf(format, v...))
	}
}
```